docs/dpdk/gnmi-ctl.rst

```diff
@@ -0,0 +1,288 @@
+to the Config monitoring service running as part of Stratum and integrated as part of
+infrap4d. This utility is built as part of networking recipe and is available under
+`install/bin/`.
+
+gnmi client connects to gRPC port 9339 opened by the server and sends a protobuf
+based message. Refer to gnmi.proto and gnmi_ext.proto for more details on type of
+messages and services that are available for a gNMI client.
+
+Each CLI format should be inline with the YANG parser available in server which
+is a tree based hierarchy. Tree has `interfaces` as root-node, followed by the
+device type. Each device can have multiple ports, where port-name acts as a KEY.
+SET command can be executed to configure the ports and GET command to retrieve the
+information about previously configured ports
+
+1) Set attributes for a vhost port::
+
+    $ gnmi-ctl set PARAMS
+    $ Example:
+    gnmi-ctl set "device:virtual-device,name:net_vhost0,device-type:VIRTIO_NET"
+    gnmi-ctl set "device:virtual-device,name:net_vhost0,port-type:LINK"
+    gnmi-ctl set "device:virtual-device,name:net_vhost0,host-name:host1,
+                  device-type:VIRTIO_NET,queues:1,
+                  socket-path:/tmp/vhost-user-0,port-type:LINK"
+    gnmi-ctl set "device:virtual-device,name:net_vhost0,host-name:host1,
+                  device-type:VIRTIO_NET,queues:1,
+                  pipeline-name:pipe,mempool-name:MEMPOOL0,mtu:2000,
+                  socket-path:/tmp/vhost-user-0,packet-dir:host,port-type:LINK"
+ ..
+
+* Parameter description:
+    * ``PARAMS``: These params are key:value pairs. Here virtual-device is a sub-node which can hold multiple ports like net_vhost0, net_vhost1,... and each port accepts multiple configuration parameters. These configuration parameters are again a key:value pair, which can be either passed in single CLI command or multiple CLI commands.
+
+    * ``name``: Supported values are defined in chassis config file `dpdk_port_config.pb.txt`.
+    * ``host-name``: can be any string.
+    * ``queues``: number of queues to be configured by target backend.
+    * ``socket-path``: socket path for vhost port.
+    * ``port-type``: Defines the port type and supported value for vhost port is `LINK`.
+    * ``device-type``: Defines the device type and supported value for vhost port is `VIRTIO_NET`.
+    * ``pipeline-name``: Defines the pipeline name, it's a non-mandatory parameter. Default value is `pipe`.
+    * ``mempool-name``: Defines mempool name, it's a non-mandatory parameter. Default value is `MEMPOOL0`.
+    * ``mtu``: Defines maximum transmission unit, it's a non-mandatory parameter. Default value is `1500`.
+    * ``packet-dir``: Accepts 2 values `host` or `network` (case insensitive). The value `host` indicates that traffic via this port will be within the host. The value `network` indicates this port will be able to send/receive traffic to/from network. It's a non-mandatory parameter and default value is `host`.
```

Comment:
I would suggest "Value may be `host` or `network` (case-insensitive)."

`packet-dir` doesn't accept _two_ values; it accepts _one of two_ values. :)